codeit/4-sprint-mission-s7/src/main/java/com/sprint/mission/discodeit/controller/UserController.java

```diff
@@ -0,0 +1,145 @@
+
+    UserDto createdUser = userService.create(userCreateRequest, profileRequest);
+
+    log.info("HTTP Create User succeeded: id={}", createdUser.id());
+
+    return ResponseEntity
+            .status(HttpStatus.CREATED)
+            .body(createdUser);
+  }
+
+  @PatchMapping(
+      path = "{userId}",
+      consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}
+  )
+  @Override
+  public ResponseEntity<UserDto> update(
+      @PathVariable("userId") UUID userId,
+      @Valid @RequestPart("userUpdateRequest") UserUpdateRequest userUpdateRequest,
+      @RequestPart(value = "profile", required = false) MultipartFile profile
+  ) {
+
+    log.info("HTTP Update User requested: id={}, profilePresent={}",
+            userId, profile != null && !profile.isEmpty());
+
+    Optional<BinaryContentCreateRequest> profileRequest = Optional.ofNullable(profile)
+        .flatMap(this::resolveProfileRequest);
+
+    UserDto updatedUser = userService.update(userId, userUpdateRequest, profileRequest);
+
+    log.info("HTTP Update User succeeded: id={}", updatedUser.id());
+
+    return ResponseEntity
+            .status(HttpStatus.OK)
+            .body(updatedUser);
+  }
+
+  @DeleteMapping(path = "{userId}")
+  @Override
+  public ResponseEntity<Void> delete(@PathVariable("userId") UUID userId) {
+
+    log.info("HTTP Delete User requested: id={}", userId);
+
+    userService.delete(userId);
+
+    log.info("HTTP Delete User succeeded: id={}", userId);
+
+    return ResponseEntity
+            .status(HttpStatus.NO_CONTENT)
+            .build();
+  }
+
+  @GetMapping
+  @Override
+  public ResponseEntity<List<UserDto>> findAll() {
+    List<UserDto> users = userService.findAll();
+    return ResponseEntity
+        .status(HttpStatus.OK)
+        .body(users);
+  }
+
+  @PatchMapping(path = "{userId}/userStatus")
+  @Override
+  public ResponseEntity<UserStatusDto> updateUserStatusByUserId(@PathVariable("userId") UUID userId,
+      @Valid @RequestBody UserStatusUpdateRequest request) {
+    UserStatusDto updatedUserStatus = userStatusService.updateByUserId(userId, request);
+    return ResponseEntity
+        .status(HttpStatus.OK)
+        .body(updatedUserStatus);
+  }
+
+  private Optional<BinaryContentCreateRequest> resolveProfileRequest(MultipartFile profileFile) {
+    if (profileFile.isEmpty()) {
+      return Optional.empty();
+    } else {
+      try {
+        BinaryContentCreateRequest binaryContentCreateRequest = new BinaryContentCreateRequest(
+            profileFile.getOriginalFilename(),
+            profileFile.getContentType(),
+            profileFile.getBytes()
+        );
+        return Optional.of(binaryContentCreateRequest);
+      } catch (IOException e) {
+
+        log.debug("Profile read failed: originalFilename='{}', contentType={}",
+                profileFile.getOriginalFilename(), profileFile.getContentType(), e);
+
+        throw new RuntimeException(e);
+      }
+    }
+  }
```

Comment:
if else 로 작성하지마시고 if하나만 작성하는게 가독성이 높을것 같습니다. 
```suggestion
    if (profileFile.isEmpty()) {
      return Optional.empty();
    }
    try {
      BinaryContentCreateRequest binaryContentCreateRequest = new BinaryContentCreateRequest(
          profileFile.getOriginalFilename(),
          profileFile.getContentType(),
          profileFile.getBytes()
      );
      return Optional.of(binaryContentCreateRequest);
    } catch (IOException e) {

      log.debug("Profile read failed: originalFilename='{}', contentType={}",
              profileFile.getOriginalFilename(), profileFile.getContentType(), e);

      throw new RuntimeException(e);
    }
  }
```